src/use-swr.ts

```diff
@@ -531,7 +531,12 @@ function useSWR<Data = any, Error = any>(
 
   // mounted (client side rendering)
   useIsomorphicLayoutEffect(() => {
-    if (!key) return undefined
+    if (!key) {
+      if (keyRef.current !== key) unmountedRef.current = false
```

Comment:
I'm not sure if we should touch this. Since when `key` is falsy this effect returns `undefined` so it doesn't have the cleanup to reset this ref to true (https://github.com/vercel/swr/blob/7fe001c4174a8e28c5e1b7d8b8b38f941cfe39bf/src/use-swr.ts#L654).

Reply:
The test looks great, thank you!